website/server/controllers/api-v3/tasks.js

```diff
@@ -312,9 +204,15 @@ api.updateTask = {
 
     let taskId = req.params.taskId;
     let task = await Tasks.Task.findByIdOrAlias(taskId, user._id);
+    let group;
 
     if (!task) {
       throw new NotFound(res.t('taskNotFound'));
+    } else if (task.group.id) {
```

Comment:
@TheHollidayInn continuing from previous discussion because it's not outdated, I see one problem here: you're passing `user._id` to `Tasks.Task.findByIdOrAlias` so a task with no `userId`(like the original ones for challenges and group) will never be found. So lines 217-219 of this file will never be reached.

As for challenges original tasks there should be some special logic for when `task.userId` doesn't exist. If task.userId exists task.challenge or task.groups should not be considered at all except for sanitization (see line 231)